datafusion/physical-plan/src/sorts/merge.rs

```diff
@@ -303,7 +311,7 @@ impl<C: CursorValues> SortPreservingMergeStream<C> {
             (Some(ac), Some(bc)) => {
                 let ord = ac.cmp(bc);
                 if ord.is_eq() {
-                    (true, true)
+                    (true, false)
                 } else {
```

Comment:
Can you double-check here? I updated here since when the `Ordering` is `Equal`, I didn't see any reason to return `is_gt` part as `true`.

Reply:
Actually we don't need `is_gt` if `is_eq` is true. But I agree return false makes more sense.